pkg/internal/executors/executor.go

```diff
@@ -49,7 +50,22 @@ type javaExecutorFactory interface {
 	NewJavaExecutor(JavaExecutorOptions) (*JavaExecutor, error)
 }
 
-func NewJavaExecutor(opts JavaExecutorOptions) (*JavaExecutor, error) {
+func NewJavaExecutor(opts JavaExecutorOptions, attachDebugger bool) (*JavaExecutor, error) {
+	if attachDebugger {
+		opts.UseExecutor = "mvnDebug"
+		e, err := combineJavaExecutorFactories(
+			&maven{},
+		).NewJavaExecutor(opts)
+		if e == nil {
+			return nil, errors.New(
+				"failed to configure executor.  For debugging only the maven executor is supported, " +
+					"and mvnDebug must be installed")
```

Comment:
When's the case that `e` is `nil` but `err` is also `nil`?

Reply:
If the project is not a maven project, `e` is nil, but and `err` also, since it's not really an error from the point of view of the maven executor (it's just not that kind of project, there was no failure to set it up).